wallet-new/test/unit/TxMetaStorageSpecs.hs

```diff
@@ -0,0 +1,167 @@
+       release = liftIO . closeMetaDB
+
+
+-- | Generates two 'TxMeta' which are @almost@ identical, if not in the
+-- arrangement of their inputs & outputs.
+genSimilarTxMetas :: IO (TxMeta, TxMeta)
+genSimilarTxMetas = liftIO $ do
+    inputs  <- generate (resize 5 arbitrary)
+    outputs <- generate (resize 5 arbitrary)
+    blueprint <- generate arbitrary
+    let t1 = blueprint & over txMetaInputs  (const inputs)
+                       . over txMetaOutputs (const outputs)
+    let t2 = blueprint & over txMetaInputs  (const (NonEmpty.reverse inputs))
+                       . over txMetaOutputs (const (NonEmpty.reverse outputs))
+    return (t1, t2)
+
+
+-- | Handy wrapper to be able to compare things with the 'isomorphicTo'
+-- combinator, which ignores the different order of the inputs & outputs.
+data DeepEqual = DeepEqual TxMeta
+
+instance Eq DeepEqual where
+    (DeepEqual t1) == (DeepEqual t2) = t1 `exactlyEqualTo` t2
+
+instance Buildable DeepEqual where
+    build (DeepEqual t) = build t
+
+data Isomorphic = Isomorphic TxMeta
+
+instance Eq Isomorphic where
+    (Isomorphic t1) == (Isomorphic t2) = t1 `isomorphicTo` t2
+
+instance Buildable Isomorphic where
+    build (Isomorphic t) = build t
+
+instance Buildable [Isomorphic] where
+    build ts = bprint (listJsonIndent 4) ts
+
+sortByAmount :: SortDirection -> [Isomorphic] -> [Isomorphic]
+sortByAmount direction = sortBy sortFn
+    where
+        withDir Ascending  = identity
+        withDir Descending = flip
+
+        sortFn (Isomorphic a) (Isomorphic b) =
+            (withDir direction compare) (a ^. txMetaAmount) (b ^. txMetaAmount)
+
+sortByCreationAt :: SortDirection -> [Isomorphic] -> [Isomorphic]
+sortByCreationAt direction = sortBy sortFn
+    where
+        withDir Ascending  = identity
+        withDir Descending = flip
+
+        sortFn (Isomorphic a) (Isomorphic b) =
+            (withDir direction compare) (a ^. txMetaCreationAt) (b ^. txMetaCreationAt)
+
+-- | Specs which tests the persistent storage and API provided by 'TxMeta'.
+txMetaStorageSpecs :: Spec
+txMetaStorageSpecs = do
+    describe "TxMeta equality" $ do
+        it "should be reflexive" $ do
+            (blueprint :: TxMeta) <- liftIO $ generate arbitrary
+            blueprint `exactlyEqualTo` blueprint `shouldBe` True
+
+        it "should be strict when needed" $ do
+            (t1, t2) <- liftIO genSimilarTxMetas
+            t1 `exactlyEqualTo` t2 `shouldBe` False
+
+        it "isomorphicTo is more lenient" $ do
+            (t1, t2) <- liftIO genSimilarTxMetas
+            t1 `isomorphicTo` t2 `shouldBe` True
+
+    describe "TxMeta storage" $ do
+
+        it "can store a TxMeta and retrieve it back" $ do
+            withTemporaryDb $ \hdl -> do
+                testMeta <- liftIO $ generate arbitrary
+                void $ putTxMeta hdl testMeta
+                mbTx <- getTxMeta hdl (testMeta ^. txMetaId)
+                fmap DeepEqual mbTx `shouldBe` Just (DeepEqual testMeta)
```

Comment:
We may want to use [`Test.QuickCheck.Monadic`](http://hackage.haskell.org/package/QuickCheck-2.11.3/docs/Test-QuickCheck-Monadic.html) here. I think this would be expressed as:

```haskell
it "can store a TxMeta and retrieve it back" $ monadicIO $ do
    testMeta <- pick
    run $ withTemporaryDb $ \hdl -> do
        void $ putTxMeta hdl testMeta
        mbTx <- getTxMeta hdl (testMeta ^. txMetaId)
        fmap DeepEqual mbTx `shouldBe` Just (DeepEqual testMeta)
```

That would generate 100 examples and try it on each of them, instead of just a single one.